classes/class-nets-easy-templates.php

```diff
@@ -53,62 +53,83 @@ public function __construct() {
 				return $template;
 			}
 
-			if ( is_wc_endpoint_url( 'order-pay' ) ) {
-				return $template;
+			$maybe_template = locate_template( 'woocommerce/nets-easy-checkout.php' );
+			$nexi_template  = $maybe_template ? $maybe_template : WC_DIBS_PATH . '/templates/nets-easy-checkout.php';
+
+			$chosen_payment_method = WC()->session->chosen_payment_method;
+			if ( 'dibs_easy' === $chosen_payment_method ) {
+				return $nexi_template;
+			}
+
+			if ( empty( $chosen_payment_method ) && 'dibs_easy' === array_key_first( $available_gateways ) ) {
+				return $nexi_template;
 			}
 
-			if ( 'checkout/form-checkout.php' === $template_name ) {
-				$available_gateways = WC()->payment_gateways()->get_available_payment_gateways();
-
-				if ( locate_template( 'woocommerce/nets-easy-checkout.php' ) ) {
-					$dibs_easy_checkout_template = locate_template( 'woocommerce/nets-easy-checkout.php' );
-				} else {
-					$dibs_easy_checkout_template = WC_DIBS_PATH . '/templates/nets-easy-checkout.php';
-				}
-
-				// DIBS Easy checkout page.
-				if ( array_key_exists( 'dibs_easy', $available_gateways ) ) {
-					// If chosen payment method exists.
-					if ( 'dibs_easy' === WC()->session->get( 'chosen_payment_method' ) ) {
-						$template = $dibs_easy_checkout_template;
-					}
-
-					// If chosen payment method does not exist and Easy is the first gateway.
-					if ( null === WC()->session->get( 'chosen_payment_method' ) || '' === WC()->session->get( 'chosen_payment_method' ) ) {
-						reset( $available_gateways );
-
-						if ( 'dibs_easy' === key( $available_gateways ) ) {
-							$template = $dibs_easy_checkout_template;
-						}
-					}
-
-					// If another gateway is saved in session, but has since become unavailable.
-					if ( WC()->session->get( 'chosen_payment_method' ) && ! array_key_exists( WC()->session->get( 'chosen_payment_method' ), $available_gateways ) ) {
-						reset( $available_gateways );
-
-						if ( 'dibs_easy' === key( $available_gateways ) ) {
-							$template = $dibs_easy_checkout_template;
-						}
-					}
-				}
+			if ( ! isset( $available_gateways[ $chosen_payment_method ] ) && 'dibs_easy' === array_key_first( $available_gateways ) ) {
+				return $nexi_template;
 			}
 		}
 
 		return $template;
 	}
 
+	/**
+	 * Replaces the payment method template only.
+	 *
+	 * @param string $template The absolute path to the template.
+	 * @param string $template_name The relative path to the template (known as the 'name').
+	 * @return string
+	 */
+	public function replace_payment_method( $template, $template_name ) {
+		if ( 'checkout/payment.php' === $template_name ) {
+			WC()->session->set( 'chosen_payment_method', 'dibs_easy' );
+			// Retrieve the template for Nexi Checkout template.
+			$maybe_template = locate_template( 'woocommerce/nets-easy-inline.php' );
+			return $maybe_template ? $maybe_template : WC_DIBS_PATH . '/templates/nets-easy-inline.php';
+
+		}
+
+		return $template;
```

Comment:
Here you set the chosen payment method always, and load our template as long as its the payment.php template. But, what if nexi isnt the already chosen option? For the checkout template we make several checks to ensure the correct template is loaded, and the correct template is loaded in the case of a fallback, like if nexi is the first option and the previously selected is no longer available. Do we not need that here too?